File: 02160.cpp
#include <iostream>
using namespace std;

int different(char pic1[35], char pic2[35]) {
	int cnt = 0;
	for (int i = 0; i < 35; i++)
		if (pic1[i] != pic2[i])
			cnt++;

	return cnt;
}

char pics[50][35];

int main() {
	int N; cin >> N;
	for (int i = 0; i < N; i++) {
		int index = 0;
		for (int j = 0; j < 5; j++)
			for (int k = 0; k < 7; k++)
				cin >> pics[i][index++];
	}

	int p1 = 0, p2 = 0, min = 36;
	for (int i = 0; i < N - 1; i++) {
		for (int j = i + 1; j < N; j++) {
			int dif = different(pics[i], pics[j]);
			if (dif < min) {
				p1 = i; p2 = j;
				min = dif;
			}
		}
	}

	cout << p1 + 1 << ' ' << p2 + 1 << endl;
}

File: 02160.py
def different(pic1, pic2):
    cnt = 0
    for i in range(35):
        if pic1[i] != pic2[i]:
            cnt += 1

    return cnt

N = int(input())
pics = []
for i in range(N):
    s = ""
    for j in range(5):
        s += input()
    pics.append(s)

p1 = p2 = 0; min = 36
for i in range(N - 1):
    for j in range(i + 1, N):
        dif = different(pics[i], pics[j])
        if dif < min:
            p1 = i; p2 = j
            min = dif

print(p1 + 1, p2 + 1)
